tests/e2e/awesome_functions/awesome_functions.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2023 Memgraph Ltd.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+# License, and you may not use this file except in compliance with the Business Source License.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+import sys
+
+import pytest
+from common import memgraph
+
+
+def test_indexed_join_with_indices(memgraph):
+    def get_bytes(prop_name):
+        res = list(memgraph.execute_and_fetch(f"MATCH (n) RETURN property_size(n, '{prop_name}') AS size"))
+        return res[0]["size"]
+
+    memgraph.execute(
+        """
+        CREATE (n:Node)
+        SET n.null_prop = null,
+            n.bool_prop = True,
+            n.S_int_prop = 4,
+            n.M_int_prop = 500,
+            n.L_int_prop = 1000000000,
+            n.XL_int_prop = 1000000000000,
+            n.float_prop = 4.0,
+            n.str_prop = 'str_value',
+            n.list_prop = [1, 2, 3],
+            n.map_prop = {key1: 'value', key2: 4},
+            n.date_prop = date('2023-01-01'),
+            n.localtime_prop = localtime('23:00:00'),
+            n.localdatetime_prop = localdatetime('2022-01-01T23:00:00'),
+            n.duration_prop = duration('P5DT2M2.33S');
+        """
+    )
+
+    null_bytes = get_bytes("null_prop")
+    bool_bytes = get_bytes("bool_prop")
+    s_int_bytes = get_bytes("S_int_prop")
+    m_int_bytes = get_bytes("M_int_prop")
+    l_int_bytes = get_bytes("L_int_prop")
+    xl_int_bytes = get_bytes("XL_int_prop")
+    float_bytes = get_bytes("float_prop")
+    str_bytes = get_bytes("str_prop")
+    list_bytes = get_bytes("list_prop")
+    map_bytes = get_bytes("map_prop")
+    date_bytes = get_bytes("date_prop")
+    local_time_bytes = get_bytes("localtime_prop")
+    local_date_time_bytes = get_bytes("localdatetime_prop")
+    duration_bytes = get_bytes("duration_prop")
+    nonexistent_bytes = get_bytes("nonexistent_prop")
+
+    # No property stored, no bytes allocated
+    assert null_bytes == 0
+
+    # 1 byte metadata, 1 byte prop id, but value is encoded in the metadata
+    assert bool_bytes == 2
+
+    # 1 byte metadata, 1 byte prop id + payload size 1 byte to store the int
+    assert s_int_bytes == 3
```

Comment:
Is this because int stored is 4 so we can use int8?

Reply:
yes, if it fits in a shorter byte int then we encode it in less bytes